### ISSUE TITLE
replace OpenRouteService with Open Source Routing Machine

### ISSUE THREAD

**Issue by jancborchardt** [libreprojects/libreprojects#83]
Apparently OpenRouteService is deprecated and not updated since 2012. The new stuff is http://map.project-osrm.org/

**Comment by airon90** [comment]
+1

**Comment by yuwash** [comment]
It's really a pity openrouteservice isn't available any more. OSRM is nicely designed, but unfortunately only has routing for cars.
- another alternative based on OSM is http://www.yournavigation.org/
- btw openmarkers seems to be discontinued too

**Comment by jancborchardt** [comment]
Right, it’s a pity OSRM only has car navigation. But the fact that it’s very well designed and seems to be the main/popular »OSM routing thingy« right now is reason enough to make it replace OpenRouteService.

I just tried http://www.yournavigation.org/ and while it has a bunch of options, you don’t really get the turn-by-turn directions. And stuff like the time or distance you only get as raw data dump on the info pane.

**Comment by yuwash** [comment]
Ok, now I noticed OpenRouteService isn't dead but just inactive.

And I agree that [YOURS](http://wiki.openstreetmap.org/wiki/YOURS) or at least the version installed at yournavigation.org is not mature enough for production use.

An indeed very active online routing project is [graphhopper/graphhopper](https://github.com/graphhopper/graphhopper/) (last commit today!). The hosted instance is at http://graphhopper.com
- This shows accurate direction with distance in km and time in minutes
- provides fast comparison links to OSRM, Google and Bing and a GPX download
- car, bicycle and pedestrian
- multiple map layers to choose from

**Comment by jancborchardt** [comment]
@yuwash yeah, somewhere on the OpenStreetMap wiki it said OpenRouteService is discontinued.

Wow, graphhopper looks really good actually! That would indeed be a good replacement for OpenRouteService. Can you change it via pull request (but with a logo please ;)

**Comment by yuwash** [comment]
As also referenced above, I've made the commit 53c75fba7e2f054fb62166c37d06979c09ae1e33 for graphhopper, but I won't request a pull before the pump.io pull is merged (I've improperly made that in the gh-pages branch, so it'll be mixed up otherwise). In the meantime you can already take a look at it and make it better.

**Comment by benediktg** [comment]
openrouteservice.org has been renewed, you might want look at it once again.

**Comment by jancborchardt** [comment]
@benediktg ah, thanks for the notice! :) But even though on the surface it looks nicer (with Bootstrap etc), the interaction is still more complicated than Graphhopper. The first fields shown to me on openrouteservice.org are »maximum speed« and dropdowns for roads I want to avoid … instead of just typing in start and destination.
